echemdb/data/cv/entry.py

```diff
@@ -175,28 +175,29 @@ def format_title(self, e, which_field, as_sentence=True):
 
     def x(self):
         r"""
-        Return the name of the variable on the x-axis, i.e., `"U"`.
-
-        TODO: Adapt along with https://github.com/echemdb/svgdigitizer/issues/106.
+        Return the name of the variable on the x-axis, i.e., ``"E"``.
 
         EXAMPLES::
 
             >>> entry = Entry.create_examples()[0]
             >>> entry.x()
-            'U'
+            'E'
 
         """
         from astropy import units as u
-        if u.Unit(self.figure_description.potential_scale.unit).is_equivalent('V'):
-            return 'U'
-        else:
-            raise ValueError(f"The variable on the x-axis is not equivalent to 'V'.")
+
+        if not self.data_description.axes:
+            raise IndexError("No axes were specified for the dataset.")
+        if 'E' not in self.data_description.axes._descriptor.keys():
+            raise KeyError("None of the axes has a variable 'E'.")
+        if not u.Unit(self.data_description.axes.E.unit).is_equivalent('V'):
+            raise ValueError("The variable on the x-axis denoted as'E', is not convertible to 'V'.")
+        
+        return 'E'
 
     def y(self):
```

Comment:
I think technically, you are missing the case here that the axis is labeled `I` but in `A / m²`.